src/librustc/middle/stability.rs

```diff
@@ -347,42 +350,44 @@ impl<'a, 'tcx> Visitor<'tcx> for MissingStabilityAnnotations<'a, 'tcx> {
             // optional. They inherit stability from their parents when unannotated.
             hir::ItemKind::Impl(.., None, _, _) | hir::ItemKind::ForeignMod(..) => {}
 
-            _ => self.check_missing_stability(i.id, i.span)
+            hir::ItemKind::Mod(..) => self.check_missing_stability(i.id, i.span, "module"),
+
+            _ => self.check_missing_stability(i.id, i.span, "node")
```

Comment:
Using `i.node.descriptive_variant()` might be better than specifying a name manually. Not sure if there is an equivalent for the other functions.